test/metabase/api/database_test.clj

```diff
@@ -1040,7 +1040,7 @@
 (deftest databases-list-include-saved-questions-tables-test-6
   (testing "GET /api/database?saved=true&include=tables"
     (testing "should work when there are no DBs that support nested queries"
-      (with-redefs [driver/database-supports? (constantly false)]
+      (with-redefs [driver.u/supports? (constantly false)]
```

Comment:
It seems like a very easy mistake to use the seemingly "more public" `database-supports?` methods in future tests, and to introduce confusing "works in isolation" CI failures. What do you think about making the former a discouraged var in kondo?

Reply:
I considered it. I'm not sure it's worth it because (a) `driver/database-supports?` is already a discouraged var in non-test namespaces ([code](https://github.com/metabase/metabase/blob/f3a9d26202403f43f24e7c08d55ab3b283f71382/.clj-kondo/config.edn#L862)) except driver namespaces, so that makes it less likely you'll redef `database-supports?` in tests and (b) there are enough places places where we implement `driver/database-supports?` for test drivers in test namespaces and it feels a little to awkward to add `#_{:clj-kondo/ignore [:discouraged-var]}` comments everywhere. What we really want is to prevent using `with-redefs` inside `driver/database-supports?` by default and provide an escape hatch. But I don't have an easy way to do that.

Reply:
Yeah, that makes sense. It's got me wondering two things:

1. How many of these spots should be using `with-dynamic-redef`.
2. If it makes sense to have a map of "forbidden" vars to redefine with our helper. It could fail with a point to what to use instead, and having a dynamic var to disable that. Magic 👍 👎 🎭

Reply:
Solved this problem with https://github.com/metabase/metabase/pull/44938